pkg/authz/check.go

```diff
@@ -6,13 +6,103 @@ import (
 	"fmt"
 
 	"golang.org/x/sync/errgroup"
-	"k8s.io/klog/v2"
 
 	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
 
 	"github.com/authzed/spicedb-kubeapi-proxy/pkg/rules"
 )
 
+// checkRelationships performs authorization checks for a slice of relationships
+// Uses single CheckPermission for one relationship, bulk CheckBulkPermissions for multiple
+func checkRelationships(ctx context.Context, client v1.PermissionsServiceClient, resolvedRels []*rules.ResolvedRel, checkType string) error {
+	if len(resolvedRels) == 0 {
+		return nil
+	}
+
+	if len(resolvedRels) == 1 {
+		// Single check - use regular CheckPermission
+		rel := resolvedRels[0]
+		req := &v1.CheckPermissionRequest{
+			Consistency: &v1.Consistency{
+				Requirement: &v1.Consistency_FullyConsistent{FullyConsistent: true},
+			},
+			Resource: &v1.ObjectReference{
+				ObjectType: rel.ResourceType,
+				ObjectId:   rel.ResourceID,
+			},
+			Permission: rel.ResourceRelation,
+			Subject: &v1.SubjectReference{
+				Object: &v1.ObjectReference{
+					ObjectType: rel.SubjectType,
+					ObjectId:   rel.SubjectID,
+				},
+				OptionalRelation: rel.SubjectRelation,
+			},
+		}
+		resp, err := client.CheckPermission(ctx, req)
+		if err != nil {
+			return err
+		}
+		if resp.Permissionship != v1.CheckPermissionResponse_PERMISSIONSHIP_HAS_PERMISSION {
+			return fmt.Errorf("%s failed for %s:%s#%s@%s:%s",
+				checkType, rel.ResourceType, rel.ResourceID, rel.ResourceRelation,
+				rel.SubjectType, rel.SubjectID)
+		}
+		return nil
+	}
+
+	// Multiple checks - use bulk check
```

Comment:
Why not just use bulk check always? It supports single permissions

Reply:
I was thinking there's more wrapping / unwrapping overhead